api/src/permissions/permissions.service.ts

```diff
@@ -1,31 +1,33 @@
 import { Uuid, DocType, AclPermission } from "../enums";
 import { GroupAclEntryDto } from "../dto/GroupAclEntryDto";
 import { GroupDto } from "../dto/GroupDto";
+import { AccessMap } from "./AccessMap";
+import { validateChangeRequest } from "./validateChangeReq";
 
 /**
  * Acl map entry used internally in Group objects
  */
 type AclMapEntry = {
-    ref: Group;
+    ref: PermissionSystem;
     types: Map<DocType, Map<AclPermission, boolean>>;
 };
 
 /**
  * Global Group Map used for permission lookups
  */
-const groupMap: Map<Uuid, Group> = new Map<Uuid, Group>();
+const groupMap: Map<Uuid, PermissionSystem> = new Map<Uuid, PermissionSystem>();
 
 /**
- * Permission system Group class
+ * Permission system class
  */
-export class Group {
+export class PermissionSystem {
     private _typePermissionGroupRequestorMap = new Map<
         DocType,
         Map<AclPermission, Map<Uuid, Map<Uuid, boolean>>>
     >();
     private _groupTypePermissionMap = new Map<Uuid, Map<DocType, Map<AclPermission, boolean>>>();
     private _aclMap = new Map<Uuid, AclMapEntry>();
-    private _childGroups = new Map<Uuid, Group>();
+    private _childGroups = new Map<Uuid, PermissionSystem>();
```

Comment:
These variables are still named "group".

Reply:
Keeping "group" naming still makes sense as the Permission System is in fact a tree structure where each node is representing a group. Added a comment in the code to explain this a bit.